### ISSUE TITLE
getitem for vector quantity fails when dtype is object

### ISSUE THREAD

**Issue by ipelupessy** [amusecode/amuse#236]
**Describe the bug**
getitem (ie subscripting) fails for vector quantity with a object data type

**To Reproduce**
```
>>> a=numpy.arange(10, dtype=object) | units.m
>>> print a
[0, 1, 2, 3, 4, 5, 6, 7, 8, 9] m
>>> print a[1:4]
[1, 2, 3] m
>>> print a[2]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/inti/code/amuse/amuse/src/amuse/units/quantities.py", line 590, in __getitem__
    if number.shape:
AttributeError: 'int' object has no attribute 'shape'
```
**Expected behavior**
should not fail, just return the requested object, this will/may be important when fixing #155

**Comment by ipelupessy** [comment]
one of the ways to fix #155 involves unit_with_specific_dtype whihc owuld trigger this...

**Comment by ipelupessy** [comment]
automagically fixed (at least the example given)...
